### ISSUE TITLE
Blocks with visual connections or directions cannot be replaced if adjacent to another block

### ISSUE THREAD

**Issue by niborbb** [IntellectualSites/FastAsyncWorldEdit#445]
# Bug report for FastAsyncWorldEdit 1.15.2

**[REQUIRED] FastAsyncWorldEdit Configuration Files:** 
https://athion.net/ISPaster/paste/view/f6585ce481e24f2ebbaf690fe24888de

**[REQUIRED] FastAsyncWorldEdit Version Number:** 
Build 192

**[REQUIRED] Spigot/Paper Version Number:** 
This server is running Paper version git-Paper-237 (MC: 1.15.2) (Implementing API version 1.15.2-R0.1-SNAPSHOT)

**[REQUIRED] Description of the problem:**
When replacing blocks that connect visually to another block (e.g. walls that connect to each other / to normal blocks) or blocks that have a direction (gates, logs, pillars, stairs), these blocks can no longer be replaced when they are placed next to a block.
Also when the block next to which the walls/fences/pillars/logs etc. were connected / placed is destroyed, they still cannot be replaced if you try to replace them immediately after that. You first have to execute another worldedit command. It does not matter which command it is or if it affects the walls/fences/pillars/logs or not (as long as it still is a walls/fences/stairs/logs and not //set air or something like this). Editing other blocks with FAWE has the same effect and you can //replace the walls/fences/pillars/logs after that just like normal.
In extension to #440, this replacement bug also occurs when connecting stairs so that they form inner/outer stairs. These will also not be replaced anymore, even if the original stair could be replaced and the weird behaviour with the destroyed blocks also applies to them.

**How to replicate**:
To ensure good readability, I will stick to the cobblestone wall example. It is completely equivalent for all other materials. For stairs, just make sure the stair you want to replace forms the inner/outer stair. For logs/pillars etc. just place them in the desired direction.
In general, if already the original block cannot be replaced, then nothing else in the below steps can be replaced, so make sure that the first step does work. Otherwise see #440 
- Place a cobblestone wall without connecting to anything and try to //replace it with e.g. stone -> the wall should be replaced
- Place a cobblestone wall next to a block with which they will connect or place a block next to such a wall. Make a new worldedit selection with the wall and try to //replace the cobble wall -> it should not be replaced. There is a chance that it will be replaced, but for me that happens only once every ~10-20 edits.
(If you place them in an existing selection area, the probability that it will be replaced is also slightly higher. But this does not happen every time, and I have no idea where to start to investigate this problem in general, so this is just a side note)
- Destroy the adjacent block and try to //replace the cobblestone wall now -> should also not work in most cases
- Execute another worldedit command as long as you do not try to replace the cobblestone wall and it still is a cobblestone wall after that (so no //set, //cut or similar, but e.g. //move, //stack, //copy have the desired effect).  Replacing any other block that will be replaced does the trick too in addition to executing (almost) every other worldedit command elsewhere but on the wall. Then try to replace the cobble wall -> should work again
(I did not test every command, thus the "almost")

**Plugins being used on the server:**
ColoredSigns*, Essentials, EssentialsAntiBuild, EssentialsChat, FastAsyncWorldEdit, LuckPerms, Multiverse-Core, Multiverse-Portals, MyCommand, Vault, VoxelSniper, WorldEdit, WorldEditSelectionVisualizer

**Checklist**:

- [x] I included all information required in the sections above
- [x] I made sure there are no duplicates of this report [(Use Search)](https://github.com/IntellectualSites/FastAsyncWorldEdit/issues?q=is%3Aissue)
- [x] I made sure I am using an up-to-date version of [FastAsyncWorldEdit for 1.15.2](https://ci.athion.net/job/FastAsyncWorldEdit-1.15/)
- [x] I made sure the bug/error is not caused by any other plugin

**Comment by aurorasmiles** [comment]
To be honest, I wasn't able to replace cobblestone_walls at all, even when not connected, while testing this. My guess is that this issue is similar to #440 and #372 where FAWE somehow as an issue with blocks that have multiple blockstates, but that's something the devs have to figure out.
For now I can only confirm that bug
